debbuild

```diff
@@ -1812,181 +1983,522 @@ sub store_value {
+  my ($L) = @_;
+  my $p = $L->tostring(1);
+  vdebug($p, 4, "lua_rpm_define");
+  if ( my ($macro,$eq,$value) = $p =~ /^(\w+(?:\([^)]*\))?)(=|\s*)(.+)$/ ) {
+    store_value('define', $macro, $value);
+  }
+  return 0;
+}
+
+
+sub lua_posix_getcwd {
+  my ($L) = @_;
+  my $r = getcwd();
+  $L->pushstring($r);
+  vdebug($r, 4, "lua_posix_getcwd");
+  return 1;
+}
+
+
+sub lua_posix_stat {
+  my ($L) = @_;
+  my $n = $L->gettop();
+  if ( $n < 2 ) {
+    $L->pushnil();
+    return 1;
+  }
+  my $f = $L->tostring(1);
+  my $p = $L->tostring(2);
+  my %c = (
+    'dev'     =>  0, 'ino'     =>  1, 'mode'    =>  2,
+    'nlink'   =>  3, 'uid'     =>  4, 'gid'     =>  5,
+    'rdev'    =>  6, 'size'    =>  7, 'atime'   =>  8,
+    'mtime'   =>  9, 'ctime'   => 10, 'blksize' => 11,
+    'blocks'  => 12
+  );
+  my @r = stat($f);
+  if ( not @r ) {
+    vdebug("$f / $p => FILE NOT FOUND", 4, "lua_posix_stat");
+    $L->pushnil();
+    return 1;
+  }
+  my $r = defined($c{$p}) ? $r[$c{$p}] : undef;
+  vdebug("$f / $p => @r ($p: ".(defined($r) ? $r : "UNKNOWN").")", 4, "lua_posix_stat");
+  if ( not defined($r) ) {
+    $L->pushnil();
+    return 1;
+  }
+  $L->pushinteger($r);
+  return 1;
+}
+
+
+sub lua_pmain {
+  my ($L)  = @_;
+  my $m = $lstk[0]->{m};
+  my $st = $L->loadstring($m);
+  if ( $st == 0 ) {
+    my $base = $L->gettop();            # function index
+    $L->pushcfunction(\&lua_traceback); # push traceback function
+    $L->insert($base);                  # put it under chunk and args
+    $st = $L->pcall(0, 0, $base);
+    $L->remove($base);                  # remove traceback function
+  }
+  if ($st && !$L->isnil(-1)) {
+    my $msg = $L->tostring(-1);
+    $msg = "(error object is not a string)" if ! defined $msg;
+    print(STDERR "Lua Error: ".$msg."\n");
+    $lstk[0]->{error} = 1;
+    $L->pop(1);
+  }
+  return 0;
+}
+
+
+sub lua_init {
+  my $lsh = lua_get('Lua::API::State->new()');
+  if ( $lsh ) {
+    my $L = $lsh->open();  # create state
+    if (! defined $L ) {
+      vdebug("Lua Error: Cannot create state: not enough memory!", 3, "lua_init");
+      print(STDERR "Lua Error: Cannot create state: not enough memory!\n");
+      return;
+    }
+    $L->openlibs;
+    $L->register("print", \&lua_print);
+    $L->register("vdebug", \&lua_vdebug);
+    my %rpm_lib = ('expand' => \&lua_rpm_expand,
+                   'define' => \&lua_rpm_define);
+    $L->register("rpm", \%rpm_lib);
+    my %posix_lib = ('getcwd' => \&lua_posix_getcwd,
+                     'stat'   => \&lua_posix_stat);
+    $L->register("posix", \%posix_lib);
+    return $L;
+  } else {
+    vdebug("Lua Error: Unable to open Lua::API::State!", 3, "lua_init");
+    print(STDERR "Lua Error: Unable to open Lua::API::State!\n");
+  }
+  return;
+}
+
+
+sub lua_macro {
+  my ($m) = @_;
+  unless ( $lua_present ) {
+    print(STDERR "WARNING: Lua module not loaded! The following macro is omitted:\n".
+                  $m."\n# end of the ommited macro\n");
```

Comment:
Additionally, when printing warnings, please use `warn` and use `_()` so that they can be localized. For example:

```
      warn _('WARNING:  Missing value for macro ').$macro.
           _(' in ')."--$opt_name"._("!  Ignoring.\n");
```